src/EditorFeatures/Test2/FindReferences/FindReferencesTests.Tuples.vb

```diff
@@ -409,6 +409,225 @@ partial class Program
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    void M(int {|Definition:left|}, int right)
+    {
+        var r = ($$[|left|], right);
+        r = (right, [|left|]);
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/52621")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestTupleNonLocal(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    int {|Definition:Property|} { get; set; }
+
+    void M(string a)
+    {
+        var r = (a.Length, $$[|Property|]);
+        r = ([|Property|], Property: [|Property|]);
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/52621")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestTupleExplicitNames(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    void M(int a, int b)
+    {
+        var t = ($$[|{|Definition:x|}|]: a, y: b);
+        t = ([|x|]: b, y: a);
+        b = t.[|x|];
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/52621")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestTupleExplicitNamesSameAsLocals01(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    void M(int x, int y)
+    {
+        var t = ($$[|{|Definition:x|}|]: x, y: y);
+        t = ([|x|]: y, y: x);
+        t = ([|x|]: x, y: y);
+        b = t.[|x|];
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/52621")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestTupleExplicitNamesSameAsLocals02(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    void M(int {|Definition:x|}, int y)
+    {
+        var t = (x: $$[|x|], y: y);
+        t = (x: y, y: [|x|]);
+        t = (x: [|x|], y: y);
+        b = t.x;
+    }
+}
+]]>
+        </Document>
+    </Project>
 </Workspace>
```

Comment:
i'm finding the tests hard to totally understand because they don't show both implicit and explicit named-props in anonymous-types, and implicit and explicit arguments for tuples.

Reply:
Check the tests below:
- `TestAnonymousTypesCSharp_PropertyAccessHoverAccessedProperty` for anonymous type properties
- `TestTupleImplicitNamesOnSameTupleType` for tuple fields